### DISCUSSION TITLE
Universal signaling server

### DISCUSSION THREAD

**Discussion by Larkooo** [libp2p/rust-libp2p#5115] [Q&A]
Would it be possible to write a universal relay/signaling server? That could support both webrtc and udp/quic for incoming connections. I might be wrong but right now it seems like you can only support one platform at a time. A webrtc server will only support connections from browsers with the webrtc transport and a relay server will only support connections from native platforms

**Comment by thomaseizinger** [comment]
You can use both but currently, you have to listen on different ports for WebRTC and QUIC. We cannot (yet) share the same UDP socket even though in theory, one can demultiplex WebRTC and QUIC packets from each other quite easily.
